### PR TITLE
fix(inductor): pass SDPA head tile counts

### PR DESCRIPTION
**Pull request by ani300**
## Summary

- pass the number of head tiles to spyre_hint instead of the desired heads per tile
- retain the existing preference of at most four heads per tile
- add regression coverage for 32-, 16-, and 14-head attention

## Context

PR #4276 introduced a heads-per-tile helper, but the tiles hint consumes a tile count. For example, 32 heads need 8 tiles to produce four heads per tile; passing 4 instead produces eight heads per tile. This affects the Ministral VLM path in torch-spyre/hf-adapters#414.

This is based on current main, including merged PR #4296. PR #4296 resolves the compile-time Granite and irregular-KV failures, but this correction is still needed for Ministral numerical accuracy: decode-3 cosine improves from 0.988989 on clean main to 0.997643, above the 0.99 test threshold.

## Validation

- focused building-block test: 2 passed
- Ministral-3-3B VLM: 5/5 token agreement and matching free-run output
- LFM2-350M: 5/5 token agreement
- Granite Vision 4.1 4B: 5/5 token agreement and matching free-run output
- pre-commit: passed

### CODE REVIEW

**Review by marnold-ibm — APPROVED** [review]
#### BLOCKERs

None.

#### SUGGESTIONs

**1. Test comment says "loaded by `import torch`" — misleading** (`tests/inductor/test_building_blocks.py:571`)

The comment reads:

```python
# The backend entry point loaded by ``import torch`` has already
# registered this module. Fetch it without importing torch_spyre here.
```

`torch_spyre._inductor.decompositions` is loaded by `import torch_spyre` at line 24 of the same test file, not by `import torch`. The `sys.modules` lookup works and is the right technique (avoids a second import that would re-run registration side-effects), but the comment misstates *why* it's guaranteed to be present. Suggest:

```python
# torch_spyre._inductor.decompositions is already loaded by the
# ``import torch_spyre`` at the top of this file. Fetch it via
# sys.modules to avoid triggering side effects from re-importing.
```

### Verdict

APPROVE — the fix is correct and well-motivated. Suggestions are minor quality items, none block merge.
